modules/effects/arrow-effects-reactor/src/main/kotlin/arrow/effects/MonoK.kt

```diff
@@ -27,6 +29,14 @@ data class MonoK<A>(val mono: Mono<A>) : MonoKOf<A>, MonoKKindedJ<A> {
   fun <B> flatMap(f: (A) -> MonoKOf<B>): MonoK<B> =
     mono.flatMap { f(it).fix().mono }.k()
 
+  fun <B> bracketCase(use: (A) -> MonoKOf<B>, release: (A, ExitCase<Throwable>) -> MonoKOf<Unit>): MonoK<B> =
+    flatMap { a ->
+      use(a).fix().mono
+        .doOnNext { release(a, ExitCase.Completed) }
```

Comment:
Check if onNext == onComplete for Mono

Reply:
Actually it's not, the correct one was `onSuccess()`, which checks the `Mono` has completed successfully. Fixed it and added a test to cover cancellation.